File: CarHomeWidget/src/main/java/com/anod/car/home/model/ShortcutIconRequestHandler.java
package com.anod.car.home.model;

import android.content.ContentResolver;
import android.content.Context;
import androidx.annotation.Nullable;

import com.squareup.picasso.Request;
import com.squareup.picasso.RequestHandler;

import java.io.IOException;

import static com.squareup.picasso.Picasso.LoadedFrom.DISK;

/**
 * @author algavris
 * @date 23/08/2016.
 */

public class ShortcutIconRequestHandler extends RequestHandler {
    private final String mAuthority;
    private final ShortcutModel mModel;

    public ShortcutIconRequestHandler(Context context) {
        mAuthority = LauncherSettings.Favorites.getContentUri(context.getPackageName()).getAuthority();
        mModel = new ShortcutModel(context);
    }

    @Override
    public boolean canHandleRequest(Request data) {
        if (ContentResolver.SCHEME_CONTENT.equals(data.uri.getScheme()))
        {
            return data.uri.getAuthority().equals(mAuthority);
        }
        return false;
    }

    @Nullable
    @Override
    public Result load(Request request, int networkPolicy) throws IOException {

        ShortcutIcon icon = mModel.loadShortcutIcon(request.uri);
        return new Result(icon.bitmap, DISK);
    }

}

File: CarHomeWidget/src/main/java/com/anod/car/home/model/ShortcutIconRequestHandler.kt
package com.anod.car.home.model

import android.content.ContentResolver
import android.content.Context

import com.squareup.picasso.Request
import com.squareup.picasso.RequestHandler

import java.io.IOException

import com.squareup.picasso.Picasso.LoadedFrom.DISK

/**
 * @author algavris
 * @date 23/08/2016.
 */

class ShortcutIconRequestHandler(context: Context) : RequestHandler() {
    private val authority: String? = LauncherSettings.Favorites.getContentUri(context.packageName).authority
    private val model: ShortcutModel = ShortcutModel(context)

    override fun canHandleRequest(data: Request): Boolean {
        return if (ContentResolver.SCHEME_CONTENT == data.uri.scheme) {
            data.uri.authority == authority
        } else false
    }

    @Throws(IOException::class)
    override fun load(request: Request, networkPolicy: Int): RequestHandler.Result? {
        val icon = model.loadShortcutIcon(request.uri)
        return RequestHandler.Result(icon!!.bitmap, DISK)
    }

}
